crest/Assets/Crest/Crest/Scripts/Shapes/FFT/FFTBaker.cs

```diff
@@ -0,0 +1,194 @@
+                var encodedTexture = stagingTexture.EncodeToEXR(Texture2D.EXRFlags.OutputAsFloat);
+                File.WriteAllBytes($"{folderName}/test_{timeIndex}.exr", encodedTexture);
+#endif
+
+                frames[timeIndex] = stagingTexture.GetRawTextureData<half>().ToArray();
+            }
+
+            var framesFlattened = frames.SelectMany(x => x).ToArray();
+            //Debug.Log($"Crest: Width: {fftWaves._resolution}, frame count: {frameCount}, slices: {lodCount}, floats per frame: {frames[0].Length}, total floats: {framesFlattened.Length}");
+
+            var bakedDataSO = ScriptableObject.CreateInstance<FFTBakedData>();
+            var framesAsFloats = framesFlattened.Select(x => (float)x);
+            bakedDataSO.Initialize(
+                loopPeriod,
+                fftWaves._resolution,
+                firstLod,
+                lodCount,
+                fftWaves.WindSpeedForFFT,
+                frames.Length,
+                new half(framesAsFloats.Min()),
+                new half(framesAsFloats.Max()),
+                framesFlattened);
+
+            if (!SaveBakedDataAsset(bakedDataSO, fftWaves.gameObject.scene.name, fftWaves.gameObject.name))
+            {
+                return null;
+            }
+
+            return bakedDataSO;
+        }
+
+        private static bool SaveBakedDataAsset(ScriptableObject bakedDataSO, string sceneName, string shapeFFTName)
+        {
+            // Default folder
+            if (string.IsNullOrEmpty(s_bakeFolder))
+            {
+                s_bakeFolder = "Assets";
+            }
+
+            // Select bake folder
+            {
+                var folderPath = EditorUtility.SaveFolderPanel("Folder for baked data", s_bakeFolder, "");
+                if (string.IsNullOrEmpty(folderPath))
+                {
+                    return false;
+                }
+                s_bakeFolder = FileUtil.GetProjectRelativePath(folderPath);
+            }
+
+            var assetPath = AssetDatabase.GenerateUniqueAssetPath($"{s_bakeFolder}/{sceneName}-{shapeFFTName}-BakedData.asset");
+            AssetDatabase.CreateAsset(bakedDataSO, assetPath);
+
+            Debug.Log($"Crest: Baked wave data to {assetPath}.", bakedDataSO);
+
+            return true;
+        }
+
+        internal static void ComputeRequiredOctaves(OceanWaveSpectrum spectrum, float minIncludedWavelength, out int smallest, out int largest)
+        {
+            smallest = largest = -1;
+
+            for (var i = 0; i < OceanWaveSpectrum.NUM_OCTAVES; i++)
+            {
+                var pow = spectrum._powerDisabled[i] ? 0f : Mathf.Pow(10f, spectrum._powerLog[i]);
+                if (pow > Mathf.Pow(10f, OceanWaveSpectrum.MIN_POWER_LOG))
+                {
+                    var minWL = Mathf.Pow(2f, OceanWaveSpectrum.SMALLEST_WL_POW_2 + i);
+                    if (2f * minWL > minIncludedWavelength && smallest == -1 && minWL >= smallest)
```

Comment:
This condition is a little confusing. Why are we checking that `smallest == -1`?

Also, should `2f * minWL > minIncludedWavelength` be inclusive (ie >= instead of >)? Or maybe it is already because of `2 *`? A comment will help here.

Reply:
you're right, and it had a clause that was totally meaningless! clarified now, lmk if it makes sense

Reply:
Makes sense. Thanks.